src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,133 @@
+import { useEffect, useState } from 'react';
+
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { getFilteredPeople } from '../utils/getFilteredPeople';
+import { getPeopleWithChildren } from '../utils/getPeopleWithChildren';
+
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { ErrorMessage } from '../components/ErrorMessage';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { NoPeopleMessage } from '../components/NoPeopleMessage';
+import { NoMatchingMessage } from '../components/NoMatchingMessage';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(false);
+
+  const { slug } = useParams();
+
+  const selectedPersonSlug = slug ? slug : '';
+
+  const canShowNoPeopleMessage = people.length === 0 && !loading;
+
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const query = searchParams.get('query') || '';
+
+  const sex = searchParams.get('sex') || '';
+
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const params = new URLSearchParams(searchParams);
+
+    params.set('query', event.target.value);
+
+    if (event.target.value === '') {
+      params.delete('query');
+    }
+
+    setSearchParams(params);
+  };
+
+  const toggleCenturies = (century: string) => {
+    const params = new URLSearchParams(searchParams);
+
+    if (centuries.includes(century)) {
+      params.delete('centuries', century);
```

Comment:
The `params.delete('centuries', century);` method is incorrect. The `delete` method does not take a second argument. You should use `params.delete('centuries')` to remove all entries for 'centuries' or handle it differently if you want to remove a specific value.